juju/juju
# Fixes lp#1710899: resources output for units with no resources.

## Description of change

Previously, when application had resources but its unit did not, Juju would display "No unit resources". This is not very helpful to users as described in the linked bug. It is a lot cleaner and clearer to say that for a particular application resource, a unit resource does not exist. This PR changes the out put of 'juju resources <unit name>' command for units where an application has the resource but unit(s) not.

Resource with no indication of revision is not considered loaded. Hence, revision "-" is displayed.
To achieve this:

* displayed, i.e. formatted, resource revision had to be changed to a string instead of int;
* the "no resources" logic for unit has been narrowed down to a check for both application and unit resource. In both are not present, the unit is deemed to have "no resources";
* Previously, unit resource revision was inherited from an application resource revision when user was not interested in seeing unit resources details. We now display unit resource revision instead. 

## QA steps

1. bootstrap
2. deploy application with resources
3. check unit resources

## Documentation changes

Possibly, if unit resources is documented for situations where unit has no resources that have been defined on application.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1710899
